src/core/tests/type_prop/bitwise_ops.hpp

```diff
@@ -0,0 +1,1020 @@
+#include "openvino/core/dimension_tracker.hpp"
+#include "openvino/op/util/attr_types.hpp"
+
+using namespace ov;
+using namespace testing;
+
+template <class T>
+class BitwiseOperator : public testing::Test {};
+
+TYPED_TEST_SUITE_P(BitwiseOperator);
+
+TYPED_TEST_P(BitwiseOperator, default_constructor_integer) {
+    auto A = std::make_shared<ov::op::v0::Parameter>(element::i32,
+                                                     PartialShape{-1, 4, 1, 6, Dimension(1, 6), Dimension(2, 6)});
+    auto B = std::make_shared<ov::op::v0::Parameter>(element::i32,
+                                                     PartialShape{-1, 1, 5, 6, Dimension(5, 8), Dimension(5, 8)});
+
+    const auto op = std::make_shared<TypeParam>();
+
+    op->set_argument(0, A);
+    op->set_argument(1, B);
+
+    auto autob = ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NONE);
+    op->set_autob(autob);
+    EXPECT_EQ(op->get_autob(), ov::op::AutoBroadcastType::NONE);
+    ASSERT_THROW(op->validate_and_infer_types(), ov::NodeValidationFailure);
+
+    autob = ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY);
+    op->set_autob(autob);
+    EXPECT_EQ(op->get_autob(), ov::op::AutoBroadcastType::NUMPY);
+
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_element_type(), ov::element::i32);
+    EXPECT_EQ(op->get_output_partial_shape(0),
+              (ov::PartialShape{-1, 4, 5, 6, ov::Dimension(5, 8), ov::Dimension(5, 6)}));
+}
+
+TYPED_TEST_P(BitwiseOperator, default_constructor_boolean) {
+    auto A = std::make_shared<ov::op::v0::Parameter>(element::boolean,
+                                                     PartialShape{-1, 4, 1, 6, Dimension(1, 6), Dimension(2, 6)});
+    auto B = std::make_shared<ov::op::v0::Parameter>(element::boolean,
+                                                     PartialShape{-1, 1, 5, 6, Dimension(5, 8), Dimension(5, 8)});
+
+    const auto op = std::make_shared<TypeParam>();
+
+    op->set_argument(0, A);
+    op->set_argument(1, B);
+
+    auto autob = ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NONE);
+    op->set_autob(autob);
+    EXPECT_EQ(op->get_autob(), ov::op::AutoBroadcastType::NONE);
+    ASSERT_THROW(op->validate_and_infer_types(), ov::NodeValidationFailure);
+
+    autob = ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY);
+    op->set_autob(autob);
+    EXPECT_EQ(op->get_autob(), ov::op::AutoBroadcastType::NUMPY);
+
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_element_type(), ov::element::boolean);
+    EXPECT_EQ(op->get_output_partial_shape(0),
+              (ov::PartialShape{-1, 4, 5, 6, ov::Dimension(5, 8), ov::Dimension(5, 6)}));
+}
+
+TYPED_TEST_P(BitwiseOperator, shape_inference_2D) {
+    auto A = std::make_shared<ov::op::v0::Parameter>(element::i32, Shape{2, 2});
+    auto B = std::make_shared<ov::op::v0::Parameter>(element::i32, Shape{2, 2});
+
+    const auto op = std::make_shared<TypeParam>(A, B);
```

Comment:
If test fixuture will be child of `TypePropOpTest` then
```suggestion
    const auto op = this->make_op(a, b);
```
Consider refactor test to use it.

Reply:
Done